app/Models/Payment.php

```
@@ -135,84 +135,85 @@ public function getCollectionPaymentByBranch($date)
-            if ($payment->pdi > 0 && !$payment->pdi_approval_no) {
-                $pdi = $payment->pdi;
+    {   
+        return DB::transaction(function () use ($request) {
+            // DUPLICATE PREVENTION: Check if payment already exists
+            $existingPayment = Payment::where([
+                'loan_account_id' => $request->input('loan_account_id'),
+                'status' => 'open'
+            ])->lockForUpdate()->first();
+            
+            if ($existingPayment) {
+                \Log::info('Duplicate payment prevented. Existing payment ID: ' . $existingPayment->payment_id);
```

PHPMD (2.15.0)

Avoid using static access to class '\Log' in method 'addPayment'. (Clean Code Rules)

(StaticAccess)